x-pack/plugins/lens/public/datatable_visualization/components/table_basic.tsx

```diff
@@ -102,11 +122,35 @@ export const DatatableComponent = (props: DatatableRenderProps) => {
   );
 
   const onEditAction = useCallback(
-    (data: LensSortAction['data'] | LensResizeAction['data'] | LensToggleAction['data']) => {
+    (
+      data:
+        | LensSortAction['data']
+        | LensResizeAction['data']
+        | LensToggleAction['data']
+        | LensPagesizeAction['data']
+    ) => {
       dispatchEvent({ name: 'edit', data });
     },
     [dispatchEvent]
   );
+
+  const onChangeItemsPerPage = useCallback(
+    (pageSize) => onEditAction({ action: 'pagesize', size: pageSize }),
+    [onEditAction]
+  );
+
+  // active page isn't persisted, so we manage this state locally
+  const onChangePage = useCallback(
```

Comment:
Is this local state in the component necessary? By dispatching the edit action it should re-run the expression and pass the new value in via props right?

Reply:
Good question. I think we still need it since this state governs which page is active, something we don't persist. In other words, the edit action is dispatched when the _page size_ is changed, but not when the _active page_ is changed.

So, it makes sense (to me) that the active page be handled within the component since 
1. it isn't passed down from the visualization state
2. no change is made to the visualization state when the active page is changed.

Let me know if I'm off-base here.

Reply:
Ah, you are completely right, sorry! I missed the page index is stored in there.